### ISSUE TITLE
casting to string via the string filter does not work when using NativeEnvironment

### ISSUE THREAD

**Issue by mkrizek** [pallets/jinja#1200]
I think the following test case should pass:
```python
def test_cast_string(env):
    t = env.from_string("{{ value|string }}")
    result = t.render(value=3)
    assert isinstance(result, str)
    assert result == "3"
```

However it fails because the string filter is applied before nodes enter `native_concat`. So although the string is returned from the filter, in `native_concat` `"3"` is run through `literal_eval` which is of course evaluated to an integer value of `3` which in the end makes the `string` filter seem to be a NOOP.

```
env = <jinja2.nativetypes.NativeEnvironment object at 0x7fa893b55d00>

    def test_cast_string(env):
        t = env.from_string("{{ value|string }}")
        result = t.render(value=3)
>       assert isinstance(result, str)
E       assert False
E        +  where False = isinstance(3, str)

tests/test_nativetypes.py:43: AssertionError
```

First thought I had was to overwrite the `string` filter for `NativeEnvironment` to add additional quotes around the value so `literal_eval` evaluates it to string.

Initially I thought this was a result of us removing `preserve_quotes` but it appears to be a different issue and the test case has never worked.

## Your Environment
* Python version: 3.8.2
* Jinja version: master

cc @jctanner

**Comment by davidism** [comment]
Just like in YAML, if the value you're rendering could potentially be a different type than a string, you need to surround it in quotes to force a string. The string filter happens at a different time in the render, there's nothing that can be changed about that.

**Comment by mkrizek** [comment]
> Just like in YAML, if the value you're rendering could potentially be a different type than a string, you need to surround it in quotes to force a string.

Which is what I normally do only this time I had to use the filter because I couldn't put the quotes around the value (there is a functionality provided in Ansible where `{{` is implicit).

The following test case made me think it should work
https://github.com/pallets/jinja/blob/6478c22f29bb85b2bae635602148d95df5e5c7ce/tests/test_nativetypes.py#L105-L109 which is source of confusion as for some values (those that `literal_eval` fails on) it does work.

Eventually, I had to work around that https://github.com/ansible/ansible/pull/69044/commits/a9b934a9798c1a1162dd7218e5b2db04c8978bf4.

> The string filter happens at a different time in the render, there's nothing that can be changed about that.

I understand. Thanks for looking at this!
